src/server/cluster/incoming_slot_migration.cc

```diff
@@ -120,8 +121,10 @@ IncomingSlotMigration::~IncomingSlotMigration() {
 }
 
 void IncomingSlotMigration::Join() {
+  // TODO add timeout
   bc_->Wait();
-  state_ = MigrationState::C_FINISHED;
+  state_.store(MigrationState::C_FINISHED);
+  keys_number_ = GetKeyCount(slots_);
```

Comment:
Is this optimization so we wont calc slots key count again or should we must do this calc here to show the correct key count?
if this is an optimization I think its not so important as we the GetKeyCount is kind of the same of dbsize calc right? just gets stats from each shard

Reply:
Ok got the answer to this question from your comment above

Reply:
Ok got the answer to this question from your comment above